include/chemfiles/files/XTCFile.hpp

```diff
@@ -0,0 +1,47 @@
+// Chemfiles, a modern library for chemistry file reading and writing
+// Copyright (C) Guillaume Fraux and contributors -- BSD license
+
+#ifndef CHEMFILES_XTC_FILE_HPP
+#define CHEMFILES_XTC_FILE_HPP
+
+#include "chemfiles/File.hpp"
+#include "xdrfile.h"
+#include "xdrfile_xtc.h"
+
+namespace chemfiles {
+
+/// Simple RAII capsule for `XDRFILE*`, handling the creation and
+/// destruction of the file as needed.
+class XTCFile final : public File {
+  public:
+    XTCFile(std::string path, File::Mode mode);
+    ~XTCFile() override;
+    XTCFile(XTCFile&&) noexcept = default;
+    XTCFile& operator=(XTCFile&&) = default;
+    XTCFile(XTCFile const&) = delete;
+    XTCFile& operator=(XTCFile const&) = delete;
+
+    unsigned long get_nframes();
```

Comment:
You can just call the function `nframes` (or even `nsteps`), and it should be const.

It is also missing documentation. Something like "Get the number of frames/steps in the file, as indicated in the file header"

Reply:
Added docs and made const.